3_javascript/javascript1/week2/fullnamefunction.js

```diff
@@ -0,0 +1,192 @@
+    return "light jacket and jeans";
+  } else if (temperature >= 10) {
+    return "sweater and pants";
+  } else {
+    return "coat and warm clothes";
+  }
+}
+
+// Example :
+const clothesToWear = suggestClothing(18);
+console.log(clothesToWear); // Logs out: "light jacket and jeans"
+//4:
+
+//   Student manager
+
+//   A coding school have contacted us to make some functionality for managing their classes. We will create functionality for adding students to a class.
+
+//   For this exercise you need to figure out how Array.push works. Learn about the concept by researching about it.
+
+//   Copy the following code into your homework js file
+
+//   const class07Students = [];
+//   function addStudentToClass(studentName) {
+//     // You write code here
+//   }
+
+//   function getNumberOfStudents() {
+//     // You write code here
+//   }
+
+const class07Students = [];
+
+function addStudentToClass(studentName) {
+  // Check empty string and is not already in the class
+  if (studentName && !class07Students.includes(studentName)) {
+    // Add student
+    class07Students.push(studentName);
+    console.log(`Student ${studentName} added to the class.`);
+  } else {
+    console.log(`Invalid student name or student already in the class.`);
+  }
+}
+
+function getNumberOfStudents() {
+  return class07Students.length;
+}
+
+addStudentToClass("sanaz");
+addStudentToClass("maryam");
+addStudentToClass("hadis"); // This should print an error message
+console.log(`Number of students in the class: ${getNumberOfStudents()}`);
+
+// addStudentToClass function
+
+// The addStudentToClass should add the studentName to the class07Students array.
+
+// But there are some other requirements:
+
+// There can be only 6 students in a class. If more students are tried to be added, log out the following: "Cannot add more students to class 07".
+// The same person cannot be added to the class. If the same person is added to the class, log out the following: 'Student Benjamin is already in the class' where Benjamin is substituted for the studentName.
+// We are very fond of our Queen in Denmark, so if the Queen is added to the class she should always
+// get a space. Even if the class has been filled out.
+// You cannot add an empty string to a class
+
+const class07Students = [];
+
+function addStudentToClass(studentName) {
+  if (class07Students.length >= 6) {
+    console.log("Cannot add more students to class 07");
+    return;
+  }
+
+  if (!studentName) {
+    console.log("Student name cannot be empty");
+    return;
+  }
+
+  if (class07Students.includes(studentName)) {
+    console.log(`Student ${studentName} is already in the class`);
+    return;
+  }
+
+  if (studentName.toLowerCase() === "queen") {
+    class07Students.push(studentName);
+    console.log("Long live the Queen! Queen added to the class.");
+  } else {
+    class07Students.push(studentName);
+    console.log(`Student ${studentName} added to the class.`);
+  }
+}
```

Comment:
If I use your code and add 6 students then the Queen, I will get this message: "Cannot add more students to class 07", whereas according to the exercise, there should always be a place for the queen, so even if there are 6 people in the class, Queen should always be able to be the 7th 😇